### PR TITLE
Add Amplitude analytics tracking

### PR DESCRIPTION
**Pull request by kcole16**
Trying out [Amplitude](https://amplitude.com). So far seems much more powerful and easy to use than GA, will add in addition to GA to start. 

Large free tier too.

### CODE REVIEW

**Comment by vgrichina** [comment]
> Trying out [Amplitude](https://amplitude.com). So far seems much more powerful and easy to use than GA, will add in addition to GA to start.
> 
> Large free tier too.

yeah, GA UX is terrible

https://heap.io this one also looks interesting as you don't have to decide what data to collect beforehand

**Comment by kcole16** [comment]
> https://heap.io this one also looks interesting as you don't have to decide what data to collect beforehand

Interesting, will take a closer look

**Comment by kcole16** [comment]
@vgrichina Heap uses CSS selectors to track events. A bit worried we'll break those pretty frequently

File: src/middleware/index.js [hunk 1]
```diff
@@ -46,6 +46,10 @@ const analyticsMiddleware = store => next => action => {
       event_label: JSON.stringify(action.type)
     })
   }
+  if (window.amplitude) {
+    console.log("Yes")
```
Comment (vgrichina) [inline src/middleware/index.js]: looks like leftover from debugging

File: src/index.html [hunk 2]
```diff
@@ -9,6 +9,33 @@
       gtag('js', new Date());
       gtag('config', '{{ WALLET_GA_MEASUREMENT_ID || "" }}');
     </script>
+    <script type="text/javascript">
```
Comment (vgrichina) [inline src/index.html]: is there anything done to prevent tracking on development / staging environment?

**Comment by vgrichina** [comment]
@kcole16 weird, I don't see any banner on https://near-wallet-pr-214.onrender.com/

**Comment by kcole16** [comment]
@vgrichina, @jakestutzman pushed back on the banner, pre-mainnet at least. I think it'll be explicit enough without being overbearing if we have this text + URL change to something like testnet.wallet.nearprotocol.com

**Comment by vgrichina** [comment]
> @vgrichina Heap uses CSS selectors to track events. A bit worried we'll break those pretty frequently

I think it happens not that frequently in practice. Also we can always add special selectors for analytics if needed.

File: src/index.html (hunk 2 above)
Comment (kcole16) [reply src/index.html]: Should I use the same template tag approach as you did with GA? Wasn't quite sure how we're handling env variables in production

Comment (vgrichina) [reply src/index.html]: same approach can work, but make sure it does disable tracking (vs crashing) when used in prod

**Pull request merged**